docs/dcp2_system_design.rst

```diff
@@ -0,0 +1,1465 @@
+redundancy was one of the reasons the design of the metadata update
+mechanism became so complicated and was never finished.«
+
+When Azul indexes an analysis subgraph it needs to associate the
+analysis files in that subgraph with the properties of the biomaterial
+metadata entities in the subgraph that contains the input sequence
+files. Without this, the analysis files wouldn't be discoverable in the
+Data Browser by, say, the species, a property of the
+``donor_organism`` entity, or the assay, a property of the
+``library_construction_protocol`` entity.
+
+Azul looks for dangling edges in an analysis subgraph, that is, entities
+that occur as an input to an analysis process but that are not contained
+in the analysis subgraph. For each dangling edge, Azul queries the
+``links`` table for subgraphs containing processes that have these
+entities as outputs. It loads all matching subgraphs, connects them to
+the analysis subgraph and repeats the process until no more dangling
+edges exist. Azul then indexes the resulting stitched subgraph as it
+would any other subgraph.
+
+»To summarize, Azul dynamically builds a self-contained subgraph that
+resembles the DCP/2 analysis bundle, but without needing to redundantly
+persist the result, thereby eliminating a complicating factor for
+metadata updates. The downside is that consumers of the raw metadata in
+TDR would also have to stitch subgraphs in order to get a complete,
+self-contained analysis subgraph. One mitigation would be for Azul to
+store the all subgraphs (stitched analysis subgraphs as well as primary
+subgraphs) verbatim in its own index and expose them for access via its
+REST API. This would eliminate not only the need for consumers to stitch
+subgraphs, but also the need to learn BigQuery and the TDR schema.«
+
+Supplementary files
+-------------------
+
+… in DCP/1 were not linked as part of ``links.json``. Their mere presence
+in a bundle associated them with the project. There are no bundles in
+DCP/2 so, the schema for ``links.json`` has been refactored to
+accommodate a new type of ``supplementary_file_link``, in
+addition to the existing ``process_link``. This new type of link
+is used to associate an entity (currently only those of type
+`project`_) with a `supplementary_file`_.
+The DSS adapter scans each bundle for supplementary files and adds links
+to ``link.json`` accordingly.
+
+.. _project: https://github.com/HumanCellAtlas/metadata-schema/blob/master/json_schema/type/project/project.json
+.. _supplementary_file: https://github.com/HumanCellAtlas/metadata-schema/blob/master/json_schema/type/file/supplementary_file.json
+
+Generic file metadata
+---------------------
+
+The bundle manifest entries in the DCP/1 datastore (DSS) contain certain
+data file properties that aren't captured anywhere else. Because of the
+absence of bundles in DCP/2, these properties are instead stored in a
+*file descriptor*, as defined by the `file_descriptor.json`_
```

Comment:
(nitpick) `file_descriptor.json` displays as a link but not as ``code`` block, perhaps not possible to do both?

Reply:
Essentially no. See https://docutils.sourceforge.io/FAQ.html#is-nested-inline-markup-possible